src/dolphin/iniFile.ts

```diff
@@ -0,0 +1,278 @@
+/**
+ * rewrite of Inifile.cpp/Inifile.h from dolphin in Typescript
+ * https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Common/IniFile.cpp
+ */
+
+import fs from "fs";
+import readline from "readline";
+
+/**
+ * The IniFile Class, contains a Section subclass
+ */
+export class IniFile {
+  private sections: Section[];
+
+  public constructor() {
+    this.sections = [];
+  }
+
+  /** Differs from IniFile.cpp via:
+   * Instead of editing keyOut and valueOut by reference, return them */
+  public parseLine(line: string): Array<string | null> {
+    let retValueOut: string | null = null;
+    let keyOut: string | null = null;
+
+    if (line === "" || line[0] === "#") {
+      return [null, null];
+    }
+
+    const firstEquals = line.indexOf("=");
+    if (firstEquals !== -1) {
+      keyOut = line.substring(0, firstEquals).replace(/\s+/g, "");
+      retValueOut = line
+        .substring(firstEquals + 1)
+        .replace(/\s+/g, "")
+        .replace(/['"]+/g, "");
+    }
+
+    return [keyOut, retValueOut];
+  }
+
+  /**Differs from IniFile.cpp by:
+   * returns section object, not pointer
+   */
+  public getSection(section_name: string): Section | undefined {
```

Comment:
I know this is ported from C++ but in this repo, we don't really use snake_case for variable names. So we should probably stick with conventions established here.

```suggestion
  public getSection(sectionName: string): Section | undefined {
```